lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,31 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception = @exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = []
+
+          while exception.respond_to?(:cause) && exception.cause
```

Comment:
I'd prefer it if we conditionally defined a method for this so we're not doing the check on every backtrace. e.g.

```Ruby
def formatted_backtrace(exception = @exception)
   backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata) + 
end
```